---
title: "Shiny app V4 - MAP3K7 loss"
output:
  html_document: default
---
```{r, echo=FALSE, message=FALSE}
source("analysis scripts.R")
```
# Genomic Analysis
## Load alteration profiles
```{r, echo = FALSE, warning = FALSE} 
primary.alteration <- read.table("primary alteration.txt", header = TRUE, check.names = FALSE, sep = "\t")
primary.LOF <- read.table("primary LOF.txt", header = TRUE, check.names = FALSE, sep = "\t")
primary.GOF <- read.table("primary GOF.txt", header = TRUE, check.names = FALSE, sep = "\t")
met.alteration <- read.table("met alteration.txt", header = TRUE, check.names = FALSE, sep = "\t")
met.LOF <- read.table("met LOF.txt", header = TRUE, check.names = FALSE, sep = "\t")
met.GOF <- read.table("met GOF.txt", header = TRUE, check.names = FALSE, sep = "\t")
TCGA.alteration <- read.table("TCGA alteration.txt", header = TRUE, check.names = FALSE, sep = "\t")
TCGA.LOF <- read.table("TCGA LOF.txt", header = TRUE, check.names = FALSE, sep = "\t")
TCGA.GOF <- read.table("TCGA GOF.txt", header = TRUE, check.names = FALSE, sep = "\t")
```
## Subset alteration profiles by genetic background
Define primary and metastatic genetic backgrounds (Generalize here). For the subtype group (ST): 
If the user chooses "LOF" as the alteration, then primary.LOF and met.LOF are the input datasets.
If the user chooses "GOF" as the alteration, then primary.GOF and met.GOF are the input datasets.
```{r}
# Primary
primary.ST <- alteration_subset(primary.LOF, c("MAP3K7", "altered"))
primary.ST.patients <- colnames(primary.ST)[2:ncol(primary.ST)]
primary.WT <- alteration_subset(primary.alteration, c("MAP3K7", "unaltered"))
primary.WT.patients <- colnames(primary.alteration)[2:ncol(primary.WT)]
# Metastatic
met.ST <- alteration_subset(met.LOF, c("MAP3K7", "altered"))
met.ST.patients <- colnames(met.ST)[2:ncol(met.ST)]
met.WT <- alteration_subset(met.alteration, c("MAP3K7", "unaltered"))
met.WT.patients <- colnames(met.alteration)[2:ncol(met.WT)]
# TCGA background for DGE
TCGA.ST <- alteration_subset(TCGA.LOF, c("MAP3K7", "altered"))
TCGA.WT <- alteration_subset(TCGA.alteration, c("MAP3K7", "unaltered"))
```
Subset primary LOF/GOF alteration profiles by genetic background
```{r}
# Primary LOF WT
primary.LOF.WT.patients <- intersect(colnames(primary.LOF)[2:ncol(primary.LOF)], primary.WT.patients)
index <- c(TRUE)
for (i in 2:ncol(primary.LOF)){
  if (sum(colnames(primary.LOF)[i] == primary.LOF.WT.patients) == 1) {index[i] = TRUE}
  if (sum(colnames(primary.LOF)[i] == primary.LOF.WT.patients) == 0) {index[i] = FALSE}
}
primary.LOF.WT <- primary.LOF[,index]
# Primary GOF WT
primary.GOF.WT.patients <- intersect(colnames(primary.GOF)[2:ncol(primary.GOF)], primary.WT.patients)
index <- c(TRUE)
for (i in 2:ncol(primary.GOF)){
  if (sum(colnames(primary.GOF)[i] == primary.GOF.WT.patients) == 1) {index[i] = TRUE}
  if (sum(colnames(primary.GOF)[i] == primary.GOF.WT.patients) == 0) {index[i] = FALSE}
}
primary.GOF.WT <- primary.GOF[,index]
primary.GOF.WT.patients <- intersect(colnames(primary.GOF)[2:ncol(primary.GOF)], primary.WT.patients)
# Primary LOF ST
primary.LOF.ST.patients <- intersect(colnames(primary.LOF)[2:ncol(primary.LOF)], primary.ST.patients)
index <- c(TRUE)
for (i in 2:ncol(primary.LOF)){
  if (sum(colnames(primary.LOF)[i] == primary.LOF.ST.patients) == 1) {index[i] = TRUE}
  if (sum(colnames(primary.LOF)[i] == primary.LOF.ST.patients) == 0) {index[i] = FALSE}
}
primary.LOF.ST <- primary.LOF[,index]
# Primary GOF ST
primary.GOF.ST.patients <- intersect(colnames(primary.GOF)[2:ncol(primary.GOF)], primary.ST.patients)
index <- c(TRUE)
for (i in 2:ncol(primary.GOF)){
  if (sum(colnames(primary.GOF)[i] == primary.GOF.ST.patients) == 1) {index[i] = TRUE}
  if (sum(colnames(primary.GOF)[i] == primary.GOF.ST.patients) == 0) {index[i] = FALSE}
}
primary.GOF.ST <- primary.GOF[,index]
primary.GOF.ST.patients <- intersect(colnames(primary.GOF)[2:ncol(primary.GOF)], primary.ST.patients)
```
Subset metastatic LOF/GOF alteration profiles by genetic background
```{r}
# Met LOF WT
met.LOF.WT.patients <- intersect(colnames(met.LOF)[2:ncol(met.LOF)], met.WT.patients)
index <- c(TRUE)
for (i in 2:ncol(met.LOF)){
  if (sum(colnames(met.LOF)[i] == met.LOF.WT.patients) == 1) {index[i] = TRUE}
  if (sum(colnames(met.LOF)[i] == met.LOF.WT.patients) == 0) {index[i] = FALSE}
}
met.LOF.WT <- met.LOF[,index]
# Met GOF WT
met.GOF.WT.patients <- intersect(colnames(met.GOF)[2:ncol(met.GOF)], met.WT.patients)
index <- c(TRUE)
for (i in 2:ncol(met.GOF)){
  if (sum(colnames(met.GOF)[i] == met.GOF.WT.patients) == 1) {index[i] = TRUE}
  if (sum(colnames(met.GOF)[i] == met.GOF.WT.patients) == 0) {index[i] = FALSE}
}
met.GOF.WT <- met.GOF[,index]
met.GOF.WT.patients <- intersect(colnames(met.GOF)[2:ncol(met.GOF)], met.WT.patients)
# Met LOF ST
met.LOF.ST.patients <- intersect(colnames(met.LOF)[2:ncol(met.LOF)], met.ST.patients)
index <- c(TRUE)
for (i in 2:ncol(met.LOF)){
  if (sum(colnames(met.LOF)[i] == met.LOF.ST.patients) == 1) {index[i] = TRUE}
  if (sum(colnames(met.LOF)[i] == met.LOF.ST.patients) == 0) {index[i] = FALSE}
}
met.LOF.ST <- met.LOF[,index]
# Met GOF ST
met.GOF.ST.patients <- intersect(colnames(met.GOF)[2:ncol(met.GOF)], met.ST.patients)
index <- c(TRUE)
for (i in 2:ncol(met.GOF)){
  if (sum(colnames(met.GOF)[i] == met.GOF.ST.patients) == 1) {index[i] = TRUE}
  if (sum(colnames(met.GOF)[i] == met.GOF.ST.patients) == 0) {index[i] = FALSE}
}
met.GOF.ST <- met.GOF[,index]
met.GOF.ST.patients <- intersect(colnames(met.GOF)[2:ncol(met.GOF)], met.ST.patients)
```
## Compute LOF/GOF alteration rates
Compute LOF alteration rates in subtype patients
```{r}
# Primary
primary.ST.LOF.frequency <- alterationRank(primary.LOF.ST)
primary.ST.LOF.frequency <- primary.ST.LOF.frequency[,c(1,5)]
colnames(primary.ST.LOF.frequency)[2] <- "LOF alteration rate: Primary ST patients"
# Metastatic
met.ST.LOF.frequency <- alterationRank(met.LOF.ST)
met.ST.LOF.frequency <- met.ST.LOF.frequency[,c(1,5)]
colnames(met.ST.LOF.frequency)[2] <- "LOF alteration rate: Metastatic ST patients"
```
Compute LOF alteration rates in WT patients
```{r}
# Primary
primary.WT.LOF.frequency <- alterationRank(primary.LOF.WT)
primary.WT.LOF.frequency <- primary.WT.LOF.frequency[,c(1,5)]
colnames(primary.WT.LOF.frequency)[2] <- "LOF alteration rate: Primary WT patients"
# Metastatic
met.WT.LOF.frequency <- alterationRank(met.LOF.WT)
met.WT.LOF.frequency <- met.WT.LOF.frequency[,c(1,5)]
colnames(met.WT.LOF.frequency)[2] <- "LOF alteration rate: Metastatic WT patients"
```
Compute GOF alteration rates in subtype patients
```{r}
# Primary
primary.ST.GOF.frequency <- alterationRank(primary.GOF.ST)
primary.ST.GOF.frequency <- primary.ST.GOF.frequency[,c(1,5)]
colnames(primary.ST.GOF.frequency)[2] <- "GOF alteration rate: Primary ST patients"
# Metastatic
met.ST.GOF.frequency <- alterationRank(met.GOF.ST)
met.ST.GOF.frequency <- met.ST.GOF.frequency[,c(1,5)]
colnames(met.ST.GOF.frequency)[2] <- "GOF alteration rate: Metastatic ST patients"
```
Compute GOF alteration rates in WT patients
```{r}
# Primary
primary.WT.GOF.frequency <- alterationRank(primary.GOF.WT)
primary.WT.GOF.frequency <- primary.WT.GOF.frequency[,c(1,5)]
colnames(primary.WT.GOF.frequency)[2] <- "GOF alteration rate: Primary WT patients"
# Metastatic
met.WT.GOF.frequency <- alterationRank(met.GOF.WT)
met.WT.GOF.frequency <- met.WT.GOF.frequency[,c(1,5)]
colnames(met.WT.GOF.frequency)[2] <- "GOF alteration rate: Metastatic WT patients"
```
## Filter subtype-specific regulators of aggressiveness
### Create data frames for filtering
LOF alterations
```{r}
LOF.alterations <- merge.data.frame(primary.WT.LOF.frequency, primary.ST.LOF.frequency)
LOF.alterations <- merge.data.frame(LOF.alterations, met.WT.LOF.frequency)
LOF.alterations <- merge.data.frame(LOF.alterations, met.ST.LOF.frequency)
```
GOF alterations
```{r}
GOF.alterations <- merge.data.frame(primary.WT.GOF.frequency, primary.ST.GOF.frequency)
GOF.alterations <- merge.data.frame(GOF.alterations, met.WT.GOF.frequency)
GOF.alterations <- merge.data.frame(GOF.alterations, met.ST.GOF.frequency)
```
### Constraint 1 (C1): Alteration rate is greater in the primary setting (gene is co-altered in primary patients)
LOF alteration rate greater in primary subtype
```{r}
index.LOF.C1 <- (LOF.alterations$`LOF alteration rate: Primary ST patients` > LOF.alterations$`LOF alteration rate: Primary WT patients`)
LOF.C1.genes <- LOF.alterations$Hugo_Symbol[index.LOF.C1]
```
GOF alteration rate greater in primary subtype
```{r}
index.GOF.C1 <- (GOF.alterations$`GOF alteration rate: Primary ST patients` > GOF.alterations$`GOF alteration rate: Primary WT patients`)
GOF.C1.genes <- GOF.alterations$Hugo_Symbol[index.GOF.C1]
```
### Constraint 2 (C2): Alteration rate in WT metastatic patients ~ alteration rate in WT primary patients (gene is not under metastatic selection pressure)
Test C2 cut-offs.
```{r, message = FALSE, warning = FALSE}
library(ggplot2)
# Test genes captured at different effect sizes and report as table
C2.table <- data.frame()
for (i in seq(.2, 1e-5, length.out = 20)){
  index.LOF.C2 <- (abs(LOF.alterations$`LOF alteration rate: Metastatic WT patients`-LOF.alterations$`LOF alteration rate: Primary WT patients`) <= i)
  LOF.C2.genes <- LOF.alterations$Hugo_Symbol[index.LOF.C2]
  index.GOF.C2 <- ((abs(GOF.alterations$`GOF alteration rate: Metastatic WT patients`- GOF.alterations$`GOF alteration rate: Primary WT patients`)) <= i)
  GOF.C2.genes <- GOF.alterations$Hugo_Symbol[index.GOF.C2]
  LOF.genes <- intersect(LOF.C1.genes, LOF.C2.genes)
  GOF.genes <- intersect(GOF.C1.genes, GOF.C2.genes)
  genes.captured <- length(LOF.genes) + length(GOF.genes)
  C2.table[nrow(C2.table)+1,c(1,2)] <- c(i, genes.captured)
}
colnames(C2.table) <- c("WT Alteration Ratio Difference: |Met - Primary|", "Genes Remaining")
library(DT)
ggplot(data = C2.table, aes(`WT Alteration Ratio Difference: |Met - Primary|`, `Genes Remaining`)) + geom_point() + xlab("Constraint 2 Cutoff")
print("A C2 cutoff of <= 2% captures ~2500 genes that remain for analysis as subtype-specific regulators of aggressiveness")
```
Set C2 cutoff
```{r}
C2.cutoff <- 0.02
```
Apply C2 cutoff
```{r}
# Filter LOF alterations
index.LOF.C2 <- (abs(LOF.alterations$`LOF alteration rate: Metastatic WT patients`-LOF.alterations$`LOF alteration rate: Primary WT patients`) <= C2.cutoff)
LOF.C2.genes <- LOF.alterations$Hugo_Symbol[index.LOF.C2]
# Filter GOF alterations
index.GOF.C2 <- ((abs(GOF.alterations$`GOF alteration rate: Metastatic WT patients`- GOF.alterations$`GOF alteration rate: Primary WT patients`)) <= C2.cutoff)
GOF.C2.genes <- GOF.alterations$Hugo_Symbol[index.GOF.C2]
```
## Report ranked alterations
Assign alteration status
```{r}
filtered.LOF.alterations <- intersect(LOF.C1.genes, LOF.C2.genes)
filtered.GOF.alterations <- intersect(GOF.C1.genes, GOF.C2.genes)
LOF.genes <- setdiff(filtered.LOF.alterations, filtered.GOF.alterations)
GOF.genes <- setdiff(filtered.GOF.alterations, filtered.LOF.alterations)
altered.genes <- c(LOF.genes, GOF.genes)
```
## Apply LOF/GOF gene filter to alteration datasets
```{r}
# Filter Primary alteration
index <- c()
for (i in 1:nrow(primary.alteration)){
  if ((sum(primary.alteration$Hugo_Symbol[i] == altered.genes) > 0)) {index[i] = TRUE} else {index[i] = FALSE}
}
primary.alteration <- primary.alteration[index,]
# Filter Primary WT
index <- c()
for (i in 1:nrow(primary.WT)){
  if ((sum(primary.WT$Hugo_Symbol[i] == altered.genes) > 0)) {index[i] = TRUE} else {index[i] = FALSE}
}
primary.WT <- primary.WT[index,]
# Filter Primary ST
index <- c()
for (i in 1:nrow(primary.ST)){
  if ((sum(primary.ST$Hugo_Symbol[i] == altered.genes) > 0)) {index[i] = TRUE} else {index[i] = FALSE}
}
primary.ST <- primary.ST[index,]
# Filter Primary LOF WT
index <- c()
for (i in 1:nrow(primary.LOF.WT)){
  if ((sum(primary.LOF.WT$Hugo_Symbol[i] == LOF.genes) > 0)) {index[i] = TRUE} else {index[i] = FALSE}
}
primary.LOF.WT <- primary.LOF.WT[index,]
# Filter Primary LOF ST
index <- c()
for (i in 1:nrow(primary.LOF.ST)){
  if ((sum(primary.LOF.ST$Hugo_Symbol[i] == LOF.genes) > 0)) {index[i] = TRUE} else {index[i] = FALSE}
}
primary.LOF.ST <- primary.LOF.ST[index,]
# Filter Primary GOF WT
index <- c()
for (i in 1:nrow(primary.GOF.WT)){
  if ((sum(primary.GOF.WT$Hugo_Symbol[i] == GOF.genes) > 0)) {index[i] = TRUE} else {index[i] = FALSE}
}
primary.GOF.WT <- primary.GOF.WT[index,]
# Filter Primary GOF ST
index <- c()
for (i in 1:nrow(primary.GOF.ST)){
  if ((sum(primary.GOF.ST$Hugo_Symbol[i] == GOF.genes) > 0)) {index[i] = TRUE} else {index[i] = FALSE}
}
primary.GOF.ST <- primary.GOF.ST[index,]
# Filter Met alteration
index <- c()
for (i in 1:nrow(met.alteration)){
  if ((sum(met.alteration$Hugo_Symbol[i] == altered.genes) > 0)) {index[i] = TRUE} else {index[i] = FALSE}
}
met.alteration <- met.alteration[index,]
# Filter Met WT
index <- c()
for (i in 1:nrow(met.WT)){
  if ((sum(met.WT$Hugo_Symbol[i] == altered.genes) > 0)) {index[i] = TRUE} else {index[i] = FALSE}
}
met.WT <- met.WT[index,]
# Filter Met ST
index <- c()
for (i in 1:nrow(met.ST)){
  if ((sum(met.ST$Hugo_Symbol[i] == altered.genes) > 0)) {index[i] = TRUE} else {index[i] = FALSE}
}
met.ST <- met.ST[index,]
# Filter Met LOF WT
index <- c()
for (i in 1:nrow(met.LOF.WT)){
  if ((sum(met.LOF.WT$Hugo_Symbol[i] == LOF.genes) > 0)) {index[i] = TRUE} else {index[i] = FALSE}
}
met.LOF.WT <- met.LOF.WT[index,]
# Filter Met LOF ST
index <- c()
for (i in 1:nrow(met.LOF.ST)){
  if ((sum(met.LOF.ST$Hugo_Symbol[i] == LOF.genes) > 0)) {index[i] = TRUE} else {index[i] = FALSE}
}
met.LOF.ST <- met.LOF.ST[index,]
# Filter Met GOF WT
index <- c()
for (i in 1:nrow(met.GOF.WT)){
  if ((sum(met.GOF.WT$Hugo_Symbol[i] == GOF.genes) > 0)) {index[i] = TRUE} else {index[i] = FALSE}
}
met.GOF.WT <- met.GOF.WT[index,]
# Filter Met GOF ST
index <- c()
for (i in 1:nrow(met.GOF.ST)){
  if ((sum(met.GOF.ST$Hugo_Symbol[i] == GOF.genes) > 0)) {index[i] = TRUE} else {index[i] = FALSE}
}
met.GOF.ST <- met.GOF.ST[index,]
```
## Compute Alteration Enrichment
Compute alteration enrichment between primary WT and primary ST (co-alteration significance)
```{r}
LOF.coalteration.enrichment <- alteration.enrichment(primary.LOF.ST, primary.LOF.WT)
GOF.coalteration.enrichment <- alteration.enrichment(primary.GOF.ST, primary.GOF.WT)
```
Compute alteration enrichment between primary ST and met ST (subtype-specific regulator of aggressiveness)
```{r}
LOF.metastatic.enrichment <- alteration.enrichment(met.LOF.ST, primary.LOF.ST)
GOF.metastatic.enrichment <- alteration.enrichment(met.GOF.ST, primary.GOF.ST)
```
Construct summary table
```{r}
# Construct LOF alteration table
if (length(LOF.genes) > 0){
  LOF.genes <- as.data.frame(LOF.genes)
  LOF.genes[,2] <- "LOF"
  for (i in (1:nrow(LOF.genes))) {
    index1 <- which(primary.WT.LOF.frequency$Hugo_Symbol == LOF.genes[i,1])
    index2 <- which(primary.ST.LOF.frequency$Hugo_Symbol == LOF.genes[i,1])
    index3 <- which(LOF.coalteration.enrichment$`Hugo Symbol` == LOF.genes[i,1])
    index4 <- which(met.ST.LOF.frequency$Hugo_Symbol == LOF.genes[i,1])
    index5 <- which(LOF.metastatic.enrichment$`Hugo Symbol` == LOF.genes[i,1])
    LOF.genes[i,3] <- (primary.ST.LOF.frequency$`LOF alteration rate: Primary ST patients`[index2] 
                       - primary.WT.LOF.frequency$`LOF alteration rate: Primary WT patients`[index1])
    LOF.genes[i,4] <- LOF.coalteration.enrichment$`Enrichment p-val`[index3]
    LOF.genes[i,5] <- (met.ST.LOF.frequency$`LOF alteration rate: Metastatic ST patients`[index4] -
                         primary.ST.LOF.frequency$`LOF alteration rate: Primary ST patients`[index2])
    LOF.genes[i,6] <- LOF.metastatic.enrichment$`Enrichment p-val`[index5]
  }
colnames(LOF.genes) <- c("Hugo Symbol", "Alteration", "Primary Co-alteration Ratio Difference", "Co-alteration Enrichment pval", "Subtype Alteration Ratio Difference (Met - Primary)", "Metastatic Enrichment pval")
}
# Construct GOF alteration table
if (length(GOF.genes) > 0){
  GOF.genes <- as.data.frame(GOF.genes)
  GOF.genes[,2] <- "GOF"
  for (i in (1:nrow(GOF.genes))) {
    index1 <- which(primary.WT.GOF.frequency$Hugo_Symbol == GOF.genes[i,1])
    index2 <- which(primary.ST.GOF.frequency$Hugo_Symbol == GOF.genes[i,1])
    index3 <- which(GOF.coalteration.enrichment$`Hugo Symbol` == GOF.genes[i,1])
    index4 <- which(met.ST.GOF.frequency$Hugo_Symbol == GOF.genes[i,1])
    index5 <- which(GOF.metastatic.enrichment$`Hugo Symbol` == GOF.genes[i,1])
    GOF.genes[i,3] <- (primary.ST.GOF.frequency$`GOF alteration rate: Primary ST patients`[index2] 
                       - primary.WT.GOF.frequency$`GOF alteration rate: Primary WT patients`[index1]) 
    GOF.genes[i,4] <- GOF.coalteration.enrichment$`Enrichment p-val`[index3]
    GOF.genes[i,5] <- (met.ST.GOF.frequency$`GOF alteration rate: Metastatic ST patients`[index4] -
                         primary.ST.GOF.frequency$`GOF alteration rate: Primary ST patients`[index2])
    GOF.genes[i,6] <- GOF.metastatic.enrichment$`Enrichment p-val`[index5]
  }
colnames(GOF.genes) <- c("Hugo Symbol", "Alteration", "Primary Co-alteration Ratio Difference", "Co-alteration Enrichment pval", "Subtype Alteration Ratio Difference (Met - Primary)", "Metastatic Enrichment pval")
}
# Combine alteration tables
if ((length(LOF.genes) > 0) & (length(GOF.genes) > 0)){
  summary <- rbind(LOF.genes, GOF.genes)
  }
if ((length(LOF.genes) > 0) & (length(GOF.genes) == 0)){
  summary <- (LOF.genes)
  }
if ((length(LOF.genes) == 0) & (length(GOF.genes) > 0)){
  summary <- (GOF.genes)
}
# Clear NAs from summary table. Rows with NAs mean the alteration did not occur in both primary/met settings.
summary <- na.omit(summary)
library(DT)
datatable(summary)
```
# Transcriptomic Analysis
## Format gene expression datasets and filter genes that lack mRNA seq measurements
```{r, message=FALSE}
# Abida RNA seq data in FPKM
Abida.mRNA <- read.table("Abida mRNA seq.txt", header = TRUE, sep = "\t", check.names = FALSE)
Abida.mRNA[,2:ncol(Abida.mRNA)] <- Abida.mRNA[,2:ncol(Abida.mRNA)] + 1
Abida.mRNA[,2:ncol(Abida.mRNA)] <- log2(Abida.mRNA[,2:ncol(Abida.mRNA)])
Abida.mRNA <- Abida.mRNA[duplicated(Abida.mRNA$`Hugo_Symbol`) == FALSE,]
# TCGA RNA seq data in RSEM
TCGA.mRNA <- read.table("TCGA mRNA seq.txt", header = TRUE, sep = "\t", check.names = FALSE)
TCGA.mRNA <- TCGA.mRNA[,c(1,3:ncol(TCGA.mRNA))]
TCGA.mRNA[,2:ncol(TCGA.mRNA)] <- TCGA.mRNA[,2:ncol(TCGA.mRNA)] + 1
TCGA.mRNA[,2:ncol(TCGA.mRNA)] <- log2(TCGA.mRNA[,2:ncol(TCGA.mRNA)])
TCGA.mRNA <- TCGA.mRNA[duplicated(TCGA.mRNA$`Hugo_Symbol`) == FALSE,]
# Take intersection of genes in mRNA datasets and primary / met alteration profiles
genes <- intersect(Abida.mRNA$Hugo_Symbol, TCGA.mRNA$Hugo_Symbol)
genes <- intersect(genes, met.alteration$Hugo_Symbol)
# Filter genes from Abida mRNA dataset that are not in metastatic alteration profile
for (i in 1:nrow(Abida.mRNA)){
  if (sum(Abida.mRNA$Hugo_Symbol[i] == met.alteration$Hugo_Symbol) > 0) {index[i] <- TRUE}
  else {index[i] <- FALSE}
}
Abida.mRNA <- Abida.mRNA[index,]
# Filter genes from TCGA mRNA dataset that are not in primary alteration profile
index <- c()
for (i in 1:nrow(TCGA.mRNA)){
  if (sum(TCGA.mRNA$Hugo_Symbol[i] == primary.alteration$Hugo_Symbol) > 0) {index[i] <- TRUE}
  else {index[i] <- FALSE}
}
TCGA.mRNA <- TCGA.mRNA[index,]
```
## Compute concordant DGE on filtered datasets
```{r}
# Compute DGE in met ST
Abida.DGE <- DGE(met.ST, Abida.mRNA)
# Filter summary table for concordant DGE
Abida.DGE <- concordant_DGE(summary, Abida.DGE)
# Compute DGE in primary ST
TCGA.DGE <- DGE(primary.ST, TCGA.mRNA)
# Filter summary table for concordant DGE
TCGA.DGE <- concordant_DGE(summary, TCGA.DGE)
```
## Filter genes that lack concordant expression
```{r}
# Filter genes lacking concordant expression measurements (>= 3 samples) in both primary and metastatic settings.
Abida.DGE.filtered <- na.omit(Abida.DGE)
TCGA.DGE.filtered <- na.omit(TCGA.DGE)
genes <- intersect(Abida.DGE.filtered$`Hugo Symbol`, TCGA.DGE.filtered$`Hugo Symbol`)
index <- c()
for (i in 1:nrow(Abida.DGE.filtered)){
  if (sum(Abida.DGE.filtered[i,1] == genes) > 0){
    index[i] <- TRUE
  }
    if (sum(Abida.DGE.filtered[i,1] == genes) == 0){
    index[i] <- FALSE
  }
}
Abida.DGE.filtered <- Abida.DGE.filtered[index,]
index <- c()
for (i in 1:nrow(TCGA.DGE.filtered)){
  if (sum(TCGA.DGE.filtered[i,1] == genes) > 0){
    index[i] <- TRUE
  }
    if (sum(TCGA.DGE.filtered[i,1] == genes) == 0){
    index[i] <- FALSE
  }
}
TCGA.DGE.filtered <- TCGA.DGE.filtered[index,]
```
## Create single table holding concordant DGE for TCGA and Abida
```{r}
concordant.DGE <- data.frame()
for (i in 1:nrow(Abida.DGE.filtered)){
  concordant.DGE[i,1] <- Abida.DGE.filtered$`Hugo Symbol`[i]
  concordant.DGE[i,2] <- Abida.DGE.filtered$Alteration[i]
  concordant.DGE[i,3] <- TCGA.DGE.filtered$`Concordant FC (altered:unaltered)`[i]
  concordant.DGE[i,4] <- TCGA.DGE.filtered$pval[i]
  concordant.DGE[i,5] <- TCGA.DGE.filtered$FDR[i]
  concordant.DGE[i,6] <- Abida.DGE.filtered$`Concordant FC (altered:unaltered)`[i]
  concordant.DGE[i,7] <- Abida.DGE.filtered$pval[i]
  concordant.DGE[i,8] <- Abida.DGE.filtered$FDR[i]
}
colnames(concordant.DGE) <- c("Hugo Symbol", "Alteration", "FC Expression in Primary Subtype (Altered:Unaltered)", "Primary Expression pval", "Primary Expression FDR", "FC Expression in Metastatic Subtype (Altered:Unaltered)", "Metastatic Expression pval", "Metastatic Expression FDR")
```
## Update summary table to include concordant gene expression.
```{r}
summary.molecular <- data.frame()
for (i in 1:nrow(concordant.DGE)){
  # Add gene and alteration type
  summary.molecular[i,1] <- concordant.DGE$`Hugo Symbol`[i]
  summary.molecular[i,2] <- concordant.DGE$Alteration[i]
  # Add genomic analysis information
  genomic.index <- which(concordant.DGE$`Hugo Symbol`[i] == summary$`Hugo Symbol`)
  summary.molecular[i,3] <- summary$`Primary Co-alteration Ratio Difference`[genomic.index]
  summary.molecular[i,4] <- summary$`Co-alteration Enrichment pval`[genomic.index]
  summary.molecular[i,5] <- summary$`Subtype Alteration Ratio Difference (Met - Primary)`[genomic.index]
  summary.molecular[i,6] <- summary$`Metastatic Enrichment pval`[genomic.index]
  # Add concordant gene expression information
  summary.molecular[i,7] <- concordant.DGE$`FC Expression in Primary Subtype (Altered:Unaltered)`[i]
  summary.molecular[i,8] <- concordant.DGE$`Primary Expression pval` [i]
  summary.molecular[i,9] <- concordant.DGE$`FC Expression in Metastatic Subtype (Altered:Unaltered)`[i]
  summary.molecular[i,10] <- concordant.DGE$`Metastatic Expression pval`[i]
}
colnames(summary.molecular) <- c("Hugo Symbol", "Alteration", "Primary Co-alteration Ratio Difference", "Co-alteration Enrichment pval", "Alteration Ratio Difference in Subtype (Metastatic - Primary)", "Subtype Metastatic Enrichment pval", "FC Expression in Primary Subtype (Altered:Unaltered)", "Primary Expression pval", "FC Expression in Metastatic Subtype (Altered:Unaltered)", "Metastatic Expression pval")
```
# Clinical Analysis
Reformat TCGA alteration profiles and TCGA mRNA data for clinical analysis
```{r}
# Filter TCGA alteration profiles for genes in summary.molecular table
index <- c()
for (i in 1:nrow(TCGA.alteration)){
  if ((sum(TCGA.alteration$Hugo_Symbol[i] == altered.genes) > 0)) {index[i] = TRUE} else {index[i] = FALSE}
}
TCGA.alteration <- TCGA.alteration[index,]
# Reload TCGA.mRNA data
TCGA.mRNA <- read.table("TCGA mRNA seq.txt", header = TRUE, sep = "\t", check.names = FALSE)
TCGA.mRNA <- TCGA.mRNA[,c(1,3:ncol(TCGA.mRNA))]
TCGA.mRNA[,2:ncol(TCGA.mRNA)] <- TCGA.mRNA[,2:ncol(TCGA.mRNA)] + 1
TCGA.mRNA[,2:ncol(TCGA.mRNA)] <- log2(TCGA.mRNA[,2:ncol(TCGA.mRNA)])
TCGA.mRNA <- TCGA.mRNA[duplicated(TCGA.mRNA$`Hugo_Symbol`) == FALSE,]
# Harmonize genes in TCGA mRNA and TCGA alteration profiles
genes <- intersect(TCGA.mRNA$Hugo_Symbol, TCGA.alteration$Hugo_Symbol)
# Filter TCGA mRNA dataset
index <- c()
for (i in 1:nrow(TCGA.mRNA)){
  if (sum(TCGA.mRNA$Hugo_Symbol[i] == genes) > 0) {index[i] <- TRUE} else {index[i] <- FALSE}
}
TCGA.mRNA <- TCGA.mRNA[index,]
# Filter TCGA WT dataset
index <- c()
for (i in 1:nrow(TCGA.WT)){
  if (sum(TCGA.WT$Hugo_Symbol[i] == genes) > 0) {index[i] <- TRUE} else {index[i] <- FALSE}
}
TCGA.WT <- TCGA.WT[index,]
TCGA.WT <- TCGA.WT[duplicated(TCGA.WT$Hugo_Symbol) == FALSE,]
# Filter TCGA ST dataset
index <- c()
for (i in 1:nrow(TCGA.ST)){
  if (sum(TCGA.ST$Hugo_Symbol[i] == genes) > 0) {index[i] <- TRUE} else {index[i] <- FALSE}
}
TCGA.ST <- TCGA.ST[index,]
TCGA.ST <- TCGA.ST[duplicated(TCGA.ST$Hugo_Symbol) == FALSE,]
```
## Tumor grade in primary WT and ST
Compute association of gene expression with tumor grade in WT and ST primary patients
```{r}
# Primary WT patients
WT.grade <- gleason(TCGA.WT)
# Primary ST patients
ST.grade <- gleason(TCGA.ST)
```
Combine WT and ST gleason data
```{r}
Grade.Enrichment <- data.frame(WT.grade$`Hugo Symbol`,
                               WT.grade$`Odds Ratio`, WT.grade$`Tumor Grade pval`,
                               ST.grade$`Odds Ratio`, ST.grade$`Tumor Grade pval`)
colnames(Grade.Enrichment) <- c("Hugo Symbol",
                                "Wild-Type Odds Ratio", "Wild-Type High Grade Enrichment pval",
                                "Subtype Odds Ratio", "Subtype High Grade Enrichment pval")
```
Add tumor grade associations to summary.molecular table
```{r}
summary.full <- summary.molecular
for (i in 1:nrow(summary.full)){
  index <- which(summary.full$`Hugo Symbol`[i] == Grade.Enrichment$`Hugo Symbol`)
  summary.full$`Wild-Type Odds Ratio`[i] <- Grade.Enrichment$`Wild-Type Odds Ratio`[index]
  summary.full$`Wild-Type High Grade Enrichment pval`[i] <- Grade.Enrichment$`Wild-Type High Grade Enrichment pval`[index]
  summary.full$`Subtype Odds Ratio`[i] <- Grade.Enrichment$`Subtype Odds Ratio`[index]
  summary.full$`Subtype High Grade Enrichment pval`[i] <- Grade.Enrichment$`Subtype High Grade Enrichment pval`[index]
}
```
## DFS in primary WT and ST
Compute association of gene expression with BCR risk in ST and WT patients
```{r}
# Primary WT patients
WT.recurrence <- survival(TCGA.WT)
# Primary ST patients
ST.recurrence <- survival(TCGA.ST)
```
Combine WT and ST survival data
```{r}
survival <- data.frame(WT.recurrence$`Hugo Symbol`,
                       WT.recurrence$`DFS median difference (High Expressers - Low Expressers`, WT.recurrence$`Risk of BCR pval`, ST.recurrence$`DFS median difference (High Expressers - Low Expressers`, ST.recurrence$`Risk of BCR pval`)
colnames(survival) <- c("Hugo Symbol",
                        "Median Wild-Type DFS Difference (High Expressers - Low Expressers)", "Wild-Type DFS pval",
                        "Median Subtype DFS Difference (High Expressers - Low Expressers", "Subtype DFS pval")
```
Add BCR risk association to summary.full table
```{r}
for (i in 1:nrow(summary.full)){
  index <- which(summary.full$`Hugo Symbol`[i] == survival$`Hugo Symbol`)
  summary.full$`Median Wild-Type DFS Difference (High Expressers - Low Expressers)`[i] <- (survival$`Median Wild-Type DFS Difference (High Expressers - Low Expressers)`[index])
summary.full$`Wild-Type DFS pval`[i] <- (survival$`Wild-Type DFS pval`[index])
summary.full$`Median Subtype DFS Difference (High Expressers - Low Expressers)`[i] <- (survival$`Median Subtype DFS Difference (High Expressers - Low Expressers`[index])
summary.full$`Subtype DFS pval`[i] <- (survival$`Subtype DFS pval`[index])
}
```
## PSA expression in primary WT and ST
Compute association of gene expression with PSA expression in WT and ST primary patients
```{r}
# Primary WT patients
WT.PSA <- PSA(TCGA.WT)
# Primary ST patients
ST.PSA <- PSA(TCGA.ST)
```
Combine WT and ST PSA data
```{r}
PSA <- data.frame(WT.PSA$`Hugo Symbol`,
                  WT.PSA$`PSA t-test statistic (High Expressers - Low Expressers`, WT.PSA$p,
                  ST.PSA$`PSA t-test statistic (High Expressers - Low Expressers`, ST.PSA$p)
colnames(PSA) <- c("Hugo Symbol", 
                   "Wild-Type PSA t-test statistic (High Expressers - Low Expressers) ", "Wild-Type PSA pval",
                   "Subtype PSA t-test statistic (High Expressers - Low Expressers) ", "Subtype PSA pval")
```
Add association with PSA expression to summary.full table
```{r}
for (i in 1:nrow(summary.full)){
  index <- which(summary.full$`Hugo Symbol`[i] == PSA$`Hugo Symbol`)
  summary.full$`Wild-Type PSA Expression t-test statistic (High Expressers - Low Expressers`[i] <- PSA$`Wild-Type PSA t-test statistic (High Expressers - Low Expressers) `[index]
  summary.full$`Wild-Type PSA Expression pval`[i] <- PSA$`Wild-Type PSA pval`[index]
  summary.full$`Subtype PSA Expression t-test statistic (High Expressers - Low Expressers`[i] <- PSA$`Subtype PSA t-test statistic (High Expressers - Low Expressers) `[index]
  summary.full$`Subtype PSA Expression pval`[i] <- PSA$`Subtype PSA pval`[index]
}
```
## Add cytogenetic band to summary table
```{r}
library(org.Hs.eg.db)
loci <- Bands(summary.full$`Hugo Symbol`)
summary.full$`Cytogenetic Band` <- loci$Band
# Clear genes lacking locus annotation
summary.full <- na.omit(summary.full)
```
# Add aggregate pvals then rank by combined FDR to finalize output
```{r}
# Set weighting scheme for combined pval
genomic <- 1/3
transcriptomic <- 1/3
clinical <- 1/3
# Add combined p-values
library(metap)
for (i in 1:nrow(summary.full)){
  pvec <- c(((genomic/2)*summary.full$`Co-alteration Enrichment pval`[i]),
            ((genomic/2)*summary.full$`Subtype Metastatic Enrichment pval`[i]),
            ((transcriptomic/2)*summary.full$`Primary Expression pval`[i]),
            ((transcriptomic/2)*summary.full$`Metastatic Expression pval`[i]),
            ((clinical/2)*summary.full$`Subtype High Grade Enrichment pval`[i]),
            ((clinical/2)*summary.full$`Subtype DFS pval`[i]))
  p.combined <- sumlog(pvec)
  summary.full$`Combined p-val`[i] <- p.combined$p
}
# Order by combined p-val
summary.full <- summary.full[order(summary.full$`Combined p-val`, decreasing = FALSE),]
# Correct to combined FDR
summary.full$`Combined FDR` <- p.adjust(summary.full$`Combined p-val`, method = "fdr")
rownames(summary.full) <- 1:nrow(summary.full)
# Add Gene Rank
summary.full$`Gene Rank` <- 1:nrow(summary.full)
```
Finalized output for MAP3K7 LOF-altered background
```{r}
library(DT)
datatable(summary.full)
```